akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/JdbcSettings.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc.internal
+
+import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import com.typesafe.config.Config
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[projection] case class JdbcSettings(config: Config) {
+
+  val schema: Option[String] =
+    Option(config.getString("offset-store.schema")).filterNot(_.trim.isEmpty)
+
+  val table: String = config.getString("offset-store.table")
+
+  val dialect = {
+    val dialectToLoad = config.getString("dialect")
+    if (dialectToLoad == "<dialect>" || dialectToLoad.trim.isEmpty)
+      throw new IllegalArgumentException(
+        s"Dialect type not set. Settings [akka.projection.jdbc.dialect] currently set to $dialectToLoad")
```

Comment:
when we have a fixed list of supported dialects, we should add it to the error message.